include/generic_sort.h

```diff
@@ -207,4 +209,24 @@ class GenericDatatypeSort : public GenericSort
   Datatype gdt;
 };
 
+class DatatypeComponentSort : public GenericSort
```

Comment:
It would be good to add a comment about the purpose of this sort (most of the classes in this file don't have a comment but it would be good to have regardless).